js/src/components/shipping-rate-section/minimum-order-card/minimum-order-form-modals/minimum-order-form-modal.js

```diff
@@ -11,57 +10,48 @@ import { noop } from 'lodash';
  * Internal dependencies
  */
 import AppModal from '.~/components/app-modal';
-import AppInputPriceControl from '.~/components/app-input-price-control/index.js';
+import AppInputPriceControl from '.~/components/app-input-price-control';
 import VerticalGapLayout from '.~/components/vertical-gap-layout';
 import SupportedCountrySelect from '.~/components/supported-country-select';
+import validateMinimumOrder from './validateMinimumOrder';
 
-const AddMinimumOrderModal = ( props ) => {
-	const { countryOptions, value, onChange = noop, onRequestClose } = props;
-	const [ dropdownVisible, setDropdownVisible ] = useState( false );
-
-	const handleValidate = ( values ) => {
-		const errors = {};
-
-		if ( values.countries.length === 0 ) {
-			errors.countries = __(
-				'Please specify at least one country.',
-				'google-listings-and-ads'
-			);
-		}
-
-		if ( ! ( values.threshold > 0 ) ) {
-			errors.price = __(
-				'The minimum order amount must be greater than 0.',
-				'google-listings-and-ads'
-			);
-		}
-
-		return errors;
-	};
+/**
+ * @typedef { import("@wordpress/components").Button } Button
+ * @typedef { import(".~/data/actions").CountryCode } CountryCode
+ * @typedef { import("../typedefs.js").MinimumOrderGroup } MinimumOrderGroup
+ */
 
-	const handleSubmitCallback = ( newValue ) => {
-		onRequestClose();
-		onChange( newValue );
-	};
+/**
+ * Minimum order modal that is wrapped in a form.
+ *
+ * Buttons can be customized by using the `renderButtons` prop.
+ * The render function will be passed `formProps`,
+ * allowing the buttons to have access to form's `isValidForm` and `handleSubmit`.
+ *
+ * @param {Object} props Props.
+ * @param {Array<CountryCode>} props.countryOptions Array of country codes options, to be used as options in SupportedCountrySelect.
+ * @param {(formProps: Object) => Array<Button>} props.renderButtons Function to render buttons for the modal. `formProps` will be passed into this render function.
+ * @param {MinimumOrderGroup} props.initialValues Initial values for the form.
+ * @param {(values: MinimumOrderGroup) => void} props.onSubmit Callback when the form is submitted, with the form value.
+ * @param {() => void} props.onRequestClose Callback to close the modal.
+ */
+const MinimumOrderFormModal = ( {
+	countryOptions,
+	renderButtons = noop,
+	initialValues,
+	onSubmit,
+	onRequestClose = noop,
```

Comment:
:nail_care: YAGNI
```suggestion
	renderButtons,
	initialValues,
	onSubmit,
	onRequestClose,
```

Reply:
In regards of https://github.com/woocommerce/google-listings-and-ads/pull/1410/files/3c8edaed1176f026a862e1b8677443b70e1ad76a#r860844885 here we do not use `onRequestClose` ourselves, but only forward it, so we may also let the actual consumer of that prop (`AppModal`) decide on defaults.

Reply:
Ahh yes, agree, generally I don't use `noop` when we just forward the function and do not call the function, think I missed this when I was changing things. Removed in [`a908f74` (#1410)](https://github.com/woocommerce/google-listings-and-ads/pull/1410/commits/a908f740845ec32fe075dc8b4bcd99992206e295).